pkg/worker/loader.go

```diff
@@ -0,0 +1,126 @@
+/*
+Copyright 2023 KubeAGI.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package worker
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	arcadiav1alpha1 "github.com/kubeagi/arcadia/api/v1alpha1"
+	"github.com/kubeagi/arcadia/pkg/datasource"
+)
+
+// ModelLoader load models for worker
+type ModelLoader interface {
+	Build(ctx context.Context, model *arcadiav1alpha1.TypedObjectReference) (any, error)
+}
+
+var _ ModelLoader = (*LoaderOSS)(nil)
+
+// LoaderOSS defines the way to load model from oss
+type LoaderOSS struct {
+	c client.Client
+
+	endpoint *arcadiav1alpha1.Endpoint
+	oss      *datasource.OSS
+}
+
+func NewLoaderOSS(ctx context.Context, c client.Client, endpoint *arcadiav1alpha1.Endpoint) (ModelLoader, error) {
+	if endpoint == nil {
+		return nil, errors.New("nil oss endpoint")
+	}
+
+	oss, err := datasource.NewOSS(ctx, c, endpoint)
+	if err != nil {
+		return nil, fmt.Errorf("failed to new oss client with %w", err)
+	}
+
+	return &LoaderOSS{
+		c:        c,
+		endpoint: endpoint,
+		oss:      oss,
+	}, nil
+}
+
+// Load nothing inner go code
+func (loader *LoaderOSS) Build(ctx context.Context, model *arcadiav1alpha1.TypedObjectReference) (any, error) {
+	if model.Namespace == nil {
+		return nil, errors.New("nil namespace in model")
+	}
+	err := loader.oss.Stat(ctx, &arcadiav1alpha1.OSS{
+		Bucket: *model.Namespace,
+		Object: fmt.Sprintf("model/%s/", model.Name),
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	var accessKeyID, secretAccessKey string
```

Comment:
**Suggestion:** Consider validating the `model` parameter in the `Build` function. Currently, the function assumes that the `model` parameter is not nil, which might lead to a nil pointer dereference if the function is called with a nil `model`.
```suggestion
func (loader *LoaderOSS) Build(ctx context.Context, model *arcadiav1alpha1.TypedObjectReference) (any, error) {
  if model == nil {
    return nil, errors.New("nil model")
  }
  if model.Namespace == nil {
    return nil, errors.New("nil namespace in model")
  }
  ...
}
```